packages/lesswrong/components/common/FlashMessages.jsx

```diff
@@ -28,21 +17,21 @@ class FlashMessages extends Component {
       const translatedMessage = this.context.intl.formatMessage({ id, defaultMessage: messageString }, properties);
       return {
         ...message,
-        message: translatedMessage,
+        message: translatedMessage || messageString,
       };
     }
   }
 
   render() {
-    let messages = this.props.messages.filter(message => message.show);
-    let messageObject = messages.length > 0 && (messages[0].id || messages[0].messageString) && this.getProperties(messages[0]);
+    const { messages, clear } = this.props
+    let messageObject = messages.length > 0 && this.getProperties(messages[0]);
     return (
       <div className="flash-messages">
         <Snackbar
-          open={!!messageObject}
+          open={messages[0] && !messages[0].hide}
           message={messageObject && messageObject.message}
-          autoHideDuration = {4000}
-          onClose={this.handleRequestClose}
+          autoHideDuration={2000}
```

Comment:
This might be too fast to read?
```suggestion
          autoHideDuration={4000}
```

Reply:
I tried it, and 2000 felt best, 4000 felt quite slow.